### PR TITLE
[silgen] Mark a bunch of functions that SILGen open-codeds as "auto-generated" and thus not user generated code.

### PR DESCRIPTION
**Pull request by gottesmm**
This ensures that we do not emit diagnostics onto these functions. I noticed we
were doing this with my work on opt-remarks when I was hitting diagnostics in
these functions in a set of subsequent commits.

The way that I found these is that I went through this file and looked for any
cases where we were never calling emitStmt on a user provided function body.

I also eliminated two places where we were setting a new DebugScope even though
preEmitFunction already does that for us.

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci test

**Comment by gottesmm** [comment]
@swift-ci test source compatibility

**Comment by gottesmm** [comment]
Doing a bigger test to make sure nothing changes (in a paranoid sort of way)

File: lib/SILGen/SILGen.cpp [hunk 1]
```diff
@@ -1284,22 +1302,22 @@ void SILGenModule::emitDestructor(ClassDecl *cd, DestructorDecl *dd) {
   if (dd->hasBody()) {
     SILDeclRef destroyer(dd, SILDeclRef::Kind::Destroyer);
     SILFunction *f = getFunction(destroyer, ForDefinition);
-    preEmitFunction(destroyer, dd, f, dd);
+    RegularLocation loc(dd);
```
Comment (jckarter) [inline lib/SILGen/SILGen.cpp:1305]: This change seems like NFC here (which would be correct). Just making sure, is that intentional?

File: lib/SILGen/SILGen.cpp [hunk 2]
```diff
@@ -1284,22 +1302,22 @@ void SILGenModule::emitDestructor(ClassDecl *cd, DestructorDecl *dd) {
...
     postEmitFunction(destroyer, f);
   }
 
   // Emit the deallocating destructor.
   {
     SILDeclRef deallocator(dd, SILDeclRef::Kind::Deallocator);
     SILFunction *f = getFunction(deallocator, ForDefinition);
-    preEmitFunction(deallocator, dd, f, dd);
+    RegularLocation loc(dd);
+    preEmitFunction(deallocator, dd, f, loc);
```
Comment (jckarter) [inline lib/SILGen/SILGen.cpp:1317]: I think the deallocating entry point can be marked auto-generated. IIRC the destroyer is the one that contains the user's `deinit` code.

**Review by jckarter — APPROVED** [review]
Looks good overall.

File: lib/SILGen/SILGen.cpp (hunk 1 above)
Comment (gottesmm) [reply lib/SILGen/SILGen.cpp:1305]: I did it to match the code all around to it.

File: lib/SILGen/SILGen.cpp (hunk 2 above)
Comment (gottesmm) [reply lib/SILGen/SILGen.cpp:1317]: I think I saw an emit in emitDeallocatingDestructor. I'll check again.

Comment (gottesmm) [reply lib/SILGen/SILGen.cpp:1317]: Your right.

**Comment by gottesmm** [comment]
@swift-ci test

**Comment by gottesmm** [comment]
@swift-ci test

**Comment by gottesmm** [comment]
@swift-ci test

**Comment by gottesmm** [comment]
@swift-ci test source compatibility

**Comment by gottesmm** [comment]
@swift-ci test source compatibility

**Comment by gottesmm** [comment]
@swift-ci test source compatibility

**Comment by gottesmm** [comment]
@swift-ci test windows platform

**Comment by gottesmm** [comment]
Release did not fail b/c of this commit. Build error.

**Pull request merged**